knimin/lib/tests/test_data_access.py

```diff
@@ -747,49 +729,90 @@ def test_create_sample_plate(self):
+        # have been tested elsewhere
+        del obs['created_on']
+        exp = {'name': 'Test plate', 'plate_type_id': pt['id'],
+               'email': 'test', 'notes': None, 'studies': [9999]}
+        self.assertEqual(obs, exp)
+
+        # Update some attributes of the plate
+        exp_date = datetime.datetime.now()
+        db.edit_sample_plate(plate_id, name='Test Plate 2',
+                             created_on=exp_date, notes='Testing notes')
+        obs = db.read_sample_plate(plate_id)
+        exp = {'name': 'Test Plate 2', 'plate_type_id': pt['id'],
+               'email': 'test', 'notes': 'Testing notes', 'studies': [9999],
+               'created_on': exp_date}
+        self.assertEqual(obs, exp)
+
+        # Raises an error when no attribute is being updated
+        with self.assertRaises(ValueError) as ctx:
+            db.edit_sample_plate(plate_id)
+        self.assertEqual(ctx.exception.message,
+                         'At least one of name, plate_type_id, email, '
+                         'created_on or notes sould be provided')
+
+        # Raises an error when the plate does not exists
+        with self.assertRaises(ValueError) as ctx:
+            db.edit_sample_plate(plate_id + 1, name='test')
+        self.assertEqual(ctx.exception.message,
+                         'Sample plate ID %s does not exist.' % (plate_id + 1))
+
+        # Raises an error when sample plate name is not unique
+        dup_id = db.create_sample_plate('Test plate', pt['id'], 'test', [9999])
+        self._clean_up_funcs.insert(
+            0, partial(db.delete_sample_plate, dup_id))
+        with self.assertRaises(ValueError) as ctx:
+            db.edit_sample_plate(plate_id, name='Test plate')
+        self.assertEqual(ctx.exception.message,
+                         "Name 'Test plate' conflicts with exisiting sample "
+                         "plate %s." % dup_id)
+
+        # Raises an error when sample plate type does not exist
+        with self.assertRaises(ValueError) as ctx:
+            db.edit_sample_plate(plate_id, plate_type_id=0)
+        self.assertEqual(ctx.exception.message,
+                         'Plate type ID 0 does not exist.')
+
+        # Raises an error when plate_type_id is provided and samples have been
+        # already plated
+        # Add samples to the study
+        samples = ['9999.Sample_1', '9999.Sample_2', '9999.Sample_3']
+        db.set_study_samples(9999, samples)
+
+        # Create the layout
+        layout = []
+        row = []
+        for i in range(pt['rows']):
+            for j in range(pt['cols']):
+                row.append({'sample_id': None, 'name': None, 'notes': None})
+            layout.append(row)
+            row = []
+        layout[0][0]['sample_id'] = samples[0]
+        layout[0][1]['sample_id'] = samples[1]
+        layout[0][2]['sample_id'] = samples[2]
+        db.write_sample_plate_layout(plate_id, layout)
```

Comment:
Are these lines (795-809) actually part of this test? I don't see an assert anywhere...